test/core/phase_executor_test.py

```diff
@@ -54,3 +55,22 @@ def test_execute_phase_with_repeat_limit_max_exceeds_default_limit(self):
         openhtf.PhaseOptions(repeat_limit=phase_descriptor.MAX_REPEAT_LIMIT),
         expected_call_count=phase_descriptor.DEFAULT_REPEAT_LIMIT + 1,
     )
+
+  def test_execute_phase_when_run_if_throws_exception(self):
+
+    def run_if_with_exception():
+      raise Exception("run_if_with_exception")
+
+    @openhtf.PhaseOptions(run_if=run_if_with_exception)
+    def phase():
+      pass
+
+    _tr = None
+    def final(tr):
+      nonlocal _tr
+      _tr = tr
+
+    test = openhtf.Test(phase)
+    test.add_output_callbacks(final)
+    test.execute()
+    self.assertEqual(_tr.outcome, test_record.Outcome.ERROR)
```

Comment:
I suggest switching from unittest.TestCase to (openhtf.util) test.TestCase, and then you can use `execute_phase_or_test` and the appropriate `assertTest` method.

Reply:
Done